File: csharp/1313.decompress-run-length-encoded-list.cs
/*
 * @lc app=leetcode id=1313 lang=csharp
 *
 * [1313] Decompress Run-Length Encoded List
 */

using System.Collections.Generic;
using System.Linq;

// @lc code=start
public class Solution {
    public int[] DecompressRLElist(int[] nums) {
        return nums
            .Select((int val, int i) => (Value: val, Index: i))
            .GroupBy(t => t.Index / 2, (int k, IEnumerable<(int Value, int Index)> pair) => Enumerable.Repeat(pair.ElementAt(1).Value, pair.First().Value))
            .SelectMany(p => p)
            .ToArray();
    }
}
// @lc code=end

File: java/1313.decompress-run-length-encoded-list.java
/*
 * @lc app=leetcode id=1313 lang=java
 *
 * [1313] Decompress Run-Length Encoded List
 */

// @lc code=start
class Solution {
    public int[] decompressRLElist(int[] nums) {
        int n = nums.length, k = 0, j = 0;
        for (int i = 0; i < n; i += 2) {
            k += nums[i];
        }
        int[] ret = new int[k];
        for (int i = 0; i < n; i += 2) {
            int f = nums[i], v = nums[i + 1];
            while (f-- > 0) {
                ret[j++] = v;
            }
        }
        return ret;
    }
}
// @lc code=end

